core/src/Cabana_DeepCopy.hpp

```diff
@@ -385,6 +404,175 @@ inline void deep_copy( Slice_t &slice,
 
 //---------------------------------------------------------------------------//
 
+/**
+ * @brief Copy a partial chunk from src into the destination array
+ *
+ * @param dst
+ * @param src
+ * @param start_from
+ * @param end_from
+ * @param start_to
+ */
+template <class DstAoSoA, class SrcAoSoA>
+inline void deep_copy_partial_src(
+    DstAoSoA &dst, const SrcAoSoA &src,
+    // const int to_index,
+    // TODO: the order of these params is questionable
+    const int from_index, const int count,
+    typename std::enable_if<( is_aosoa<DstAoSoA>::value &&
+                              is_aosoa<SrcAoSoA>::value )>::type * = 0 )
+{
+    // TODO: this assumes you're trying to cross exec spaces (i.e partial copy
+    // from CPU to GPU). You can likely do this faster and avoid data
+    // duplication if that is not true
+    // TODO: it might make sense to quick path this if start=0 and end=n
+
+    // When trying to copy partial data across execution spaces using Kokkos,
+    // you have to invoke a (potentially overly) complex pattern. You are not
+    // allowed to cross execution spaces by copying subviews, so you have to
+    // take great care (and pain!) to martial you data correctly. This
+    // typically means you have to make local copies of the data of the correct
+    // size, and copy those around. In this function, that will mean making an
+    // AoSoA of type SrcAoSoA that is of reduced size, populating it, and then
+    // invoking a Cabana::depp_copy (defined above). In the future there may be
+    // a more elegant way to do this, but for now this is "safe". Typically
+    // this creates extra memory overhead as we duplicate the copyable chunk in
+    // the src memory space, which can be painful if we're using with src=GPU,
+    // but luckily that is not the most common use-case.
+
+    // Make AoSoA in src space to copy over
+
+    assert( ( size_t )( from_index + count ) <= src.size() );
+
+#ifdef CABANA_DEEPCOPY_SAFE
```

Comment:
The `ifdefs` are there because I'm not actually 100% sure what I do in the else is safe (nudging pointers along in the call to `asosoa.data()`). They may actually be the cause of the segfault the CI finds but I don't get locally..